src/typescript-service.ts

```diff
@@ -1607,6 +1614,10 @@ export class TypeScriptService {
         await this.projectManager.ensureReferencedFiles(uri).toPromise()
         this.projectManager.didSave(uri)
     }
+    
+    public publishDiagnostics() {
```

Comment:
The name `publishDiagnostics` is confusing since there already is a method `_publishDiagnostics`.  Please give either method a more specific name, such as naming this `publishDiagnosticsForOpenFiles`, or the other `publishDiagnosticsForFile`